core/api/src/mill/api/Logger.scala

```diff
@@ -29,25 +29,37 @@ trait Logger {
    * Prints miscellaneous logging output which isn't part of the main output
    * a user is looking for, but useful to provide context on what Mill is doing
    */
-  def info(s: String): Unit
+  final def info(s: String): Unit =
+    info(logKey, s)
+
+  def info(logKey: Seq[String], s: String): Unit
 
```

Comment:
Do we need these changes to the `Logger` interface?

`Logger` is shared throughout Mill as part of `core.api`, and thus it really needs to be as simple as possible. Rather than adding a more verbose version of `info`/`debug`/etc., could we instead instantiate sub-classes of `Logger` inside `MillBuildServer` that have the correct `logKey` set?

Reply:
Could we even use the `PrefixLogger` class for this? Adding a prefix to log lines is exactly what that class is meant to do

Reply:
Will try that now, but I was under the impression that `PrefixLogger` doesn't allow what I wanted here